src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -50,8 +54,13 @@ export function handleFileContent(store, content, source) {
     } else if (isGpx(content)) {
         const gpxParser = new GPX()
         const metadata = gpxParser.readMetadata(content)
+        const parseGpx = new DOMParser().parseFromString(content, 'text/xml')
         layer = new GPXLayer(source, true, 1.0, content, metadata)
-        // TODO : zoom to extent
+        const extent = getExtentForProjection(
+            store.state.position.projection,
+            bbox(gpxToGeoJSON(parseGpx))
+        )
+        store.dispatch('zoomToExtent', extent)
```

Comment:
I would also add the `!extent` check and throw `OutOfBoundsError` error here like in KML so we handle import of GPX tracks that are totally outside of CH. This is also a use case that I tested in my kml e2e tests